native/cocos/bindings/manual/jsb_platform_apple.mm

```diff
@@ -145,12 +95,14 @@ static bool JSB_loadFont(se::State &s) {
         }
 
         if (succeed) {
-            const auto &familyNamesAfterRegister = getAvailableFontFamilyNames();
-            ccstd::string familyName = getFontFamilyByCompareAvailableFontFamilyNames(familyNamesBeforeRegister, familyNamesAfterRegister);
+            CFStringRef fontName = CGFontCopyFullName(font);
+            ccstd::string familyName([(NSString *)fontName UTF8String]);
```

Comment:
Also, is the familyName the same as old approach ?

Reply:
The previous implementation uses `getFontFamilyByCompareAvailableFontFamilyNames()` to find the newly added FamilyName of `CTFontManagerRegisterGraphicsFont(font, &error)`. `CGFontCopyFullName` can achieve the same purpose and has the following advantages:
1. It will not cause related errors in special cases: the previous implementation cannot recognize fonts with similar names such as `Open Sans Ragular/Open Sans Italic/Open Sans Bold/Open Sans BoldItalic` are passed in.(They will be recognize as Open Sans and run wrong logic in getFontFamilyByCompareAvailableFontFamilyNames(): afterLen == beforeLen)
![image](https://user-images.githubusercontent.com/18634782/196392346-81cd8170-853e-4e79-8a22-90223c4039cd.png)
3. No memory leak: For the case of 1, the key-value pair cannot be set correctly for Open Sans Ragular/Open Sans Italic/Open Sans Bold/Open Sans BoldItalic to _fontFamilyNameMap. So when restart engine, CTFontManagerRegisterGraphicsFont() will be repeatedly called to cause leaks
2. More efficient: the complexity of getFontFamilyByCompareAvailableFontFamilyNames() is O(n*n)

Reply:
> Yes, Verified.

Reply:
> Also, is the familyName the same as old approach ? If it's the same, it's good for me of this change since it use less code to get the same result.

Yes, Verified.

Reply:
That's cool. Great work. 👍